libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -43,6 +43,10 @@ type
 
   RPCHandler* = proc(peer: PubSubPeer, msg: seq[RPCMsg]): Future[void] {.gcsafe.}
 
+proc hash*(p: PubSubPeer): Hash = 
+  # int is either 32/64, so intptr basically, pubsubpeer is a ref
+  cast[int](p).hash
```

Comment:
This is generally unsafe IMO, I know that the chances of getting the same type of object on the exact same location as a previous one is very low, but it could happen, moreover, if we use an object pool/mem pool those chances become very real.

Reply:
in nim this is fine, but I would indeed not do it anyway - ie might as well take the first bytes of the peerid which are equally unique

Reply:
changed it to `[pointer`] (no diff but who knows one day) and added a proper equality check in case of collision based on peerid